ad7606: add PWM and AXI clock-gen support for driver

commodo opened the pull request:
## Pull Request Description

This is just a preliminary change that can go in before others, to add AXI/SPI-engine support to the driver.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

buha commented on drivers/adc/ad7606/ad7606.h:
```diff
@@ -318,12 +327,18 @@ struct ad7606_dev {
 struct ad7606_init_param {
 	/** SPI initialization parameters */
 	struct no_os_spi_init_param spi_init;
+#ifdef XILINX_PLATFORM
```
i think we should not pollute the device driver code with #ifdefs that distinguish hardware platforms, device driver shouldn't care much about what hardware it runs on

in this case it just needs to know that it should or should not use an axi_clkgen (which could be intel, could be xilinx etc.). i suggest to remove all the ifdefs and operate on the fact that clkgen_init is populated or not, this way you move the #ifdef logic to the app level code, which indeed should care about what exact hw platform we're running on

commodo replied:
we can do it like that as well;
this was copied from `ad7616.c`, which doesn't mean that we should use that model;

i'll probably need to re-consider how to group them;
the only benefit these `#ifdef` guards have, is that there is a clear separation between "simple" mode for this driver and FPGA-mode;
the initial version of this driver, seems to have been developed as directly connected to an SoC;

commodo commented:
v1 -> v2:
* moved `struct ad7606_dev` struct into ad7606.c driver file; this makes it private now;
* split `struct ad7606_axi_dev`, `struct ad7606_axi_init_param ` and functions  `ad7606_axi_init()` & `ad7606_axi_remove()` ; these should kick-in, if there's an AXI init param struct attached to `struct ad7606_init_param`

buha commented on drivers/adc/ad7606/ad7606.c:
```diff
@@ -1197,6 +1199,10 @@ static int32_t ad7606_axi_init(struct ad7606_dev *device,
 	if (ret != 0)
 		goto error;
 
+	ret = no_os_pwm_init(&axi->trigger_pwm_desc, axi_init->trigger_pwm_init);
```
i think this makes the PWM usage mandatory while at the same time there is a convst gpio in the init_param / device driver desc 

it's very good that you added the pwm method but there should be a proper way for the user to use either method without being forced to use a particular one

buha replied:
it makes sense then

commodo replied:
> also, i think (but i'd need to check with the HDL guys) in the HDL design, the CONVST pin is wired to the CONVST pin to drive the conversion;

typo when writing here; apologies;
CONVST pin is wired to the PWM/HDL pin to drive the conversion;
